src/main/java/com/datadoghq/sketch/ddsketch/encoding/GrowingByteArrayOutput.java

```diff
@@ -0,0 +1,88 @@
+/* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2021 Datadog, Inc.
+ */
+
+package com.datadoghq.sketch.ddsketch.encoding;
+
+/**
+ * An implementation of {@link Output} that is backed by an array of bytes, whose capacity is grown
+ * as necessary.
+ */
+public final class GrowingByteArrayOutput implements Output {
+
+  private static final int INITIAL_CAPACITY = 8;
+
+  private byte[] array;
+  private int pos = 0; // invariant: pos <= array.length
+
+  public GrowingByteArrayOutput(int initialCapacity) {
+    if (initialCapacity < 0) {
+      throw new IllegalArgumentException("Capacity cannot be negative");
+    }
+    this.array = new byte[initialCapacity];
+  }
+
+  public GrowingByteArrayOutput() {
+    this(INITIAL_CAPACITY);
+  }
+
+  private void grow(int requiredCapacity) {
+    final int newCapacity = Math.max(requiredCapacity, array.length << 1);
+    final byte[] newArray = new byte[newCapacity];
+    System.arraycopy(array, 0, newArray, 0, array.length);
+    array = newArray;
+  }
+
+  @Override
+  public final void writeByte(byte value) {
+    if (pos == array.length) {
+      grow(pos + 1);
+    }
+    array[pos++] = value;
+  }
+
+  @Override
+  public final void writeLongLE(long value) {
+    if (pos > array.length - 8) {
+      grow(pos + 8);
+    }
+    array[pos] = (byte) value;
+    array[pos + 1] = (byte) (value >> 8);
+    array[pos + 2] = (byte) (value >> 16);
+    array[pos + 3] = (byte) (value >> 24);
+    array[pos + 4] = (byte) (value >> 32);
+    array[pos + 5] = (byte) (value >> 40);
+    array[pos + 6] = (byte) (value >> 48);
+    array[pos + 7] = (byte) (value >> 56);
```

Comment:
I believe these should be logical not arithmetic shifts. So `array[pos + 7] = (byte) ((value >>> 56) & 0xFF);`

Reply:
🤔 I'm not sure I see the difference that it makes, given that as far as I understand, it should only affect higher-order bits that are not kept after the conversion.

>A narrowing conversion of a signed integer to an integral type T simply discards all but the n lowest order bits, where n is the number of bits used to represent type T.

https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.1.3